src/main/scala/com/raphtory/core/components/graphbuilder/GraphBuilder.scala

```diff
@@ -9,6 +9,66 @@ import com.typesafe.config.Config
 import scala.collection.mutable
 import scala.collection.mutable.ArrayBuffer
 
+/**
+  * {s}`GraphBuilder`
+  *
+  * : `GraphBuilder` trait for creating a Graph by adding and deleting vertices and edges.
+  *    While using `GraphBuilder`, override {s}`parseTuple(tuple: T)` to define parsing of rows and use methods {s}`addVertex`, {s}`addEdge`
+  *    for adding vertices and edge to the graph. Deletion can be performed with {s}`deleteEdge` and {s}`deleteVertex` methods.
+  *
+  *    `GraphBuilder` creates a list of updates {s}`GraphUpdate` representing objects signifying vertex/edge add/delete
+  *     along with graph alteration information like update timestamp
+  *     These updates are retrieved and processed by the `Writer` and stored into `GraphPartition` as `avroSchema`
+  *     at the time of ingesting vertices, edges and processing updates.
+  *
+  * ## Methods
+  *
+  *    {s}`getUpdates(tuple: T)(failOnError: Boolean = true): List[GraphUpdate]`
+  *      : Parses `tuple` and fetches list of updates for the graph {s}`GraphUpdate`
```

Comment:
This is used internally to retrieve the updates. You should make it clear that users are not normally expected to override this!